### PR TITLE
Zaki uniswap lp interaction

### PR DESCRIPTION
**Pull request by zmanian**
WIP liquidity withdrawl

### CODE REVIEW

File: solidity/how-to-setup-hardhat.txt
Comment (jtremback) [inline solidity/how-to-setup-hardhat.txt:2]: I forgot about this file lol. I think we can remove it completely. I was just using it because it used to be confusing to set up buidlr projects. Once it's set up, of course, you can just install it with `npm install`

File: solidity/test/uniswap-logic.ts
Comment (jtremback) [inline solidity/test/uniswap-logic.ts:189]: make sure to remove these .only's before merging.

File: solidity/tests_mainnet_fork/uniswap-logic.ts
Comment (jtremback) [inline solidity/tests_mainnet_fork/uniswap-logic.ts:32]: Inconsistent indentation

File: solidity/tests_mainnet_fork/uniswap-logic.ts
Comment (jtremback) [inline solidity/tests_mainnet_fork/uniswap-logic.ts:94]: Comment above this code is inaccurate

File: solidity/tests_mainnet_fork/uniswap-logic.ts
Comment (jtremback) [inline solidity/tests_mainnet_fork/uniswap-logic.ts:171]: At least add a couple basic assertions to see that the underlying tokens have gone into the signer's account

File: solidity/contracts/TestUniswapLiquidity.sol
Comment (jtremback) [inline solidity/contracts/TestUniswapLiquidity.sol:12]: Let's use 4 space indentation

**Review by jtremback — CHANGES_REQUESTED** [review]
Some minor issues that i have put in the comments. Also, it would be good to put some very basic directions on how to run the forking tests.

File: solidity/tests_mainnet_fork/uniswap-logic.ts
Comment (jkilpatr) [reply solidity/tests_mainnet_fork/uniswap-logic.ts:101]: what exactly is it doing then? I'm trying to fix the assertions. But as far as I can tell the whole batch might not be executing.

**Review by jkilpatr — CHANGES_REQUESTED** [review]
This can't be merged until the assertions work, at the moment while the test 'succeeds' I'm pretty sure it's not running correctly.

**Comment by zmanian** [comment]
Working on asserts.

**Review by jkilpatr — APPROVED** [review]

**Comment by zmanian** [comment]
don't merge till I get the asserts in

**Comment by zmanian** [comment]
I want to Squash and merge this.

**Pull request merged**
